### PR TITLE
Document PodTopologySpread edge cases in Known Limitations

### PR DESCRIPTION
**Pull request by mrvarmazyar**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

This PR adds documentation for an important edge case in PodTopologySpread behavior to the "Known limitations" section of the topology spread constraints documentation.

**Ghost pods (pods not matching their own constraint)**
- Documents what happens when a pod's labels don't match its spread constraint's labelSelector
- Explains that such pods don't count themselves in spread calculations, allowing multiple pods to accumulate on the same topology
- Clarifies that the spreading constraint works in an unintended way rather than being ineffective

This edge case was identified while working on the scheduler codebase and was previously only documented in test files. Adding it to
user-facing documentation helps users:
- Understand unexpected scheduling behaviour
- Avoid common misconfigurations
- Debug PodTopologySpread issues more effectively

This PR complements kubernetes/kubernetes#134373, which adds similar documentation to test files.

#### Which issue(s) this PR fixes:

None - this is a documentation improvement based on internal TODO items in the scheduler test code.

#### Does this PR introduce a user-facing change?

```release-note
Document edge case in PodTopologySpread Known Limitations: pods not matching their own constraint selector ("ghost pods").
```

#### Additional documentation:

- Related test documentation PR: https://github.com/kubernetes/kubernetes/pull/134373
- Feedback from @Huang-Wei suggesting website documentation instead of only test comments

### CODE REVIEW

**Comment by reylejano** [comment]
Preview: 
https://deploy-preview-52738--kubernetes-io-main-staging.netlify.app/docs/concepts/scheduling-eviction/topology-spread-constraints/

I suggest to remove the bold to adhere to the style guide

File: content/en/docs/concepts/scheduling-eviction/topology-spread-constraints.md [hunk 1]
```diff
@@ -635,6 +635,22 @@ section of the enhancement proposal about Pod topology spread constraints.
   You can work around this by using a Node autoscaler that is aware of
   Pod topology spread constraints and is also aware of the overall set of topology
   domains.
+- **Node label typos can cause unexpected scheduling behavior.** If a node has a typo
```
Comment (reylejano) [inline content/en/docs/concepts/scheduling-eviction/topology-spread-constraints.md]: Remove bold to adhere to the style guide
```suggestion
- Node label typos can cause unexpected scheduling behavior. If a node has a typo
```

File: content/en/docs/concepts/scheduling-eviction/topology-spread-constraints.md [hunk 2]
```diff
@@ -635,6 +635,22 @@ section of the enhancement proposal about Pod topology spread constraints.
...
+  in its topology label (for example, `zon` instead of `zone`), the scheduler treats
+  that node as not having the topology key at all. This can result in:
+  - Pods failing to schedule with `UnschedulableAndUnresolvable` errors
+  - Uneven pod distribution that appears to violate spread constraints
+
+  Always verify that node labels exactly match the `topologyKey` specified in your
+  spread constraints.
+- **Pods that don't match their own labelSelector create "ghost pods".** If a pod's
```
Comment (reylejano) [inline content/en/docs/concepts/scheduling-eviction/topology-spread-constraints.md]: Remove bold to adhere to the style guide
```suggestion
- Pods that don't match their own labelSelector create "ghost pods". If a pod's
```

**Comment by mrvarmazyar** [comment]
@reylejano, thanks for the review. The bolds are removed 💯

**Comment by divya-mohan0209** [comment]
@kubernetes/sig-scheduling-approvers : PTAL &. sign off from the technical perspective

File: content/en/docs/concepts/scheduling-eviction/topology-spread-constraints.md [hunk 3]
```diff
@@ -635,6 +635,22 @@ section of the enhancement proposal about Pod topology spread constraints.
...
   Pod topology spread constraints and is also aware of the overall set of topology
   domains.
+- Node label typos can cause unexpected scheduling behavior. If a node has a typo
+  in its topology label (for example, `zon` instead of `zone`), the scheduler treats
+  that node as not having the topology key at all. This can result in:
+  - Pods failing to schedule with `UnschedulableAndUnresolvable` errors
+  - Uneven pod distribution that appears to violate spread constraints
+
+  Always verify that node labels exactly match the `topologyKey` specified in your
+  spread constraints.
```
Comment (sanposhiho) [inline content/en/docs/concepts/scheduling-eviction/topology-spread-constraints.md]: Is it really worth mentioning on the doc? Typo could lead to any unintended behavior everywhere in scheduling (or actually anything in Kubernetes).

File: content/en/docs/concepts/scheduling-eviction/topology-spread-constraints.md [hunk 4]
```diff
@@ -635,6 +635,22 @@ section of the enhancement proposal about Pod topology spread constraints.
...
+- Pods that don't match their own labelSelector create "ghost pods". If a pod's
+  labels don't match the `labelSelector` in its topology spread constraint, the pod
+  won't count itself in spread calculations. This means:
+  - Multiple such pods can accumulate on the same node without triggering the constraint
```
Comment (sanposhiho) [inline content/en/docs/concepts/scheduling-eviction/topology-spread-constraints.md]: I'd like to explain like this. You can change wordings as you like:

```suggestion
  - Multiple such pods can just accumulate on the same topology (until matching pods are newly created/deleted) because those pod's schedule don't change a spreading calculation result.
```

File: content/en/docs/concepts/scheduling-eviction/topology-spread-constraints.md [hunk 5]
```diff
@@ -635,6 +635,22 @@ section of the enhancement proposal about Pod topology spread constraints.
...
+  - The spreading constraint becomes ineffective
```
Comment (sanposhiho) [inline content/en/docs/concepts/scheduling-eviction/topology-spread-constraints.md]: What does it mean? It's actually effective (but most likely they are effective in a unintended way, in typical use cases)

File: content/en/docs/concepts/scheduling-eviction/topology-spread-constraints.md (hunk 3 above)
Comment (mrvarmazyar) [reply content/en/docs/concepts/scheduling-eviction/topology-spread-constraints.md]: @sanposhiho, You're right that typos are a general issue. Would you prefer I remove this section entirely, or keep it but phrase it differently to focus on why it's particularly non-obvious in PodTopologySpread?

File: content/en/docs/concepts/scheduling-eviction/topology-spread-constraints.md (hunk 5 above)
Comment (mrvarmazyar) [reply content/en/docs/concepts/scheduling-eviction/topology-spread-constraints.md]: Good point! Ineffective is misleading. I'll change it to "works in an unintended way".

File: content/en/docs/concepts/scheduling-eviction/topology-spread-constraints.md (hunk 3 above)
Comment (sanposhiho) [reply content/en/docs/concepts/scheduling-eviction/topology-spread-constraints.md]: I prefer removing this section

Comment (mrvarmazyar) [reply content/en/docs/concepts/scheduling-eviction/topology-spread-constraints.md]: @sanposhiho Just removed it, thanks

**Review by dipesh-rawat — COMMENTED** [review]
/approve
